<launch>

    <env name="ROS_PYTHON_LOG_CONFIG_FILE" value="$(find rhbp_core)/launch/python_logging.conf"/>

    <arg name="prefix" default="" /> <!-- RHBP manager prefix-->

    <node pkg="rhbp_core" type="planner_node.py" name="behaviour_planner_node" output="screen">
        <param name="prefix" type="string" value="$(arg prefix)" />
        <param name="automatic_stepping" type="bool" value="true"/>
        <param name="frequency" type="double" value="1.0" />
        <param name="guarantee_decision" type="bool" value="false"/> <!--Always try to come to a decision in a step-->
    </node>

    <!-- RHBP manager configuration is separated from the node
         because it uses a special namespace 'prefix/rhbp_manager'
         Limits of parameters which are accessible in dynamic_reconfigure are set in cfg/Manager.cfg
         -->
    <group ns="$(arg prefix)/rhbp_manager">
        <param name="situationBias" type="double" value="0.7" />
        <param name="predecessorBias" type="double" value="1.0" />
        <param name="successorBias" type="double" value="1.0" />
        <param name="conflictorBias" type="double" value="1.0" />
        <param name="goalBias" type="double" value="0.7" />
        <param name="planBias" type="double" value="0.3" />
        <param name="activationThreshold" type="double" value="7.0" />
        <param name="activationGoalPriority" type="bool" value="false"/>
        <param name="createLogFiles" type="bool" value="true"/>
        <param name="activation_algorithm" type="string" value="default"/>
        <param name="max_parallel_behaviours" type="int" value="-1"/> <!--Comment or negative for unlimited-->
        <param name="plan_monitoring_all_sensor_changes_by_behaviours" type="bool" value="true"/>
        <param name="plan_monitoring_behaviour_missing_influence" type="bool" value="true"/>
        <param name="plan_monitoring_unexpected_behaviour_finished" type="bool" value="true"/>
    </group>

</launch>
